### PR TITLE
build(docker): Push Docker images to Docker Hub

### PR DESCRIPTION
**Pull request by BYK**
image

### CODE REVIEW

File: docker/cloudbuild.yaml [hunk 1]
```diff
@@ -24,26 +24,40 @@ steps:
     # TODO(byk): We may also build this part as a builder image everytime
     #            there's a push to onpremise and use that image for
     curl -L "https://github.com/getsentry/onpremise/archive/v10.tar.gz" | tar xzf - --strip-components=1
-    ./install.sh
-    docker-compose run --rm web createuser --superuser --email test@example.com --password test123TEST
     # The following trick is from https://stackoverflow.com/a/52400857/90297 with great gratuity
```
Comment (BYK) [inline docker/cloudbuild.yaml]: Moving this trick above prevents the recreation of containers after the installation. Ideally, it'd be a separate PR but since it is small, I snucked it in.

File: docker/cloudbuild.yaml [hunk 2]
```diff
@@ -24,26 +24,41 @@ steps:
...
-      docker push us.gcr.io/$PROJECT_ID/sentry:latest
+      docker tag us.gcr.io/$PROJECT_ID/sentry:$COMMIT_SHA getsentry/sentry:latest
+      docker push getsentry/sentry:latest
     fi
 timeout: 2400s
 options:
   # We need more memory for Webpack builds & e2e onpremise tests
   machineType: 'N1_HIGHCPU_8'
+secrets:
+- kmsKeyName: projects/sentryio/locations/global/keyRings/service-credentials/cryptoKeys/cloudbuild
```
Comment (mattrobenolt) [inline docker/cloudbuild.yaml]: Nice.

**Review by mattrobenolt — APPROVED** [review]
